confuse/cache.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, List
+
+from . import templates
+from .core import ROOT_NAME, Configuration, ConfigView, RootView, Subview
+
+
+class CachedHandle(object):
+    """Handle for a cached value computed by applying a template on the view.
+    """
+    # some sentinel objects
+    _INVALID = object()
+    _MISSING = object()
+
+    def __init__(self, view: ConfigView, template=templates.REQUIRED) -> None:
+        self.value = self._INVALID
+        self.view = view
+        self.template = template
+
+    def get(self):
+        """Retreive the cached value from the handle.
+
+        Will re-compute the value using `view.get(template)` if it has been
+        invalidated.
+
+        May raise a `NotFoundError` if the underlying view has been
+        invalidated.
```

Comment:
Or, more pertinently, if the underlying data no longer has a value for this view?

Reply:
Updated